ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java

```diff
@@ -56,14 +60,16 @@ public Check getCheck() {
 
     // I need to make some marshalers happy ;)
     Check() {
-      this(null, 0);
+      http = null;
+      interval = null;
+      timeout = null;
     }
 
     public Check(final String url, final int intervalSec) {
-      this.http = Preconditions.checkNotNull(url, "url must not be null");
+      http = Preconditions.checkNotNull(url, "url must not be null");
       Preconditions.checkArgument(0 < intervalSec, "intervalSec must be greater than zero");
-      this.interval = intervalSec + "s";
-      this.timeout = interval;
```

Comment:
You like using this, and you removed it?
The reason I use this.xxx=xxx in c'tors is that it prevents bugs like setting a variable to itself.